src/main/kotlin/ch/bfh/habits/controllers/AuthController.kt

```
@@ -19,34 +19,53 @@ import java.util.*
-    fun login(@RequestBody body: LoginDTO): ResponseEntity<JwtTokenDTO> {
-         try {
-             authenticationManager.authenticate(UsernamePasswordAuthenticationToken(body.userName, body.password))
-         } catch (e: AuthenticationException) {
-             throw UnauthorizedException("Invalid username/password supplied")
-         }
+    fun login(@RequestBody body: LoginDTO?): ResponseEntity<Any?> {
+        try {
+            authenticationManager.authenticate(
+                UsernamePasswordAuthenticationToken(body?.userName, body?.password)
+            )
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)